src/java/org/apache/cassandra/utils/IntervalTree.java

```diff
@@ -172,51 +352,85 @@ public IntervalNode(Collection<I> toBisect)
                 intersectsRight = l;
                 left = null;
                 right = null;
+                return;
+            }
+
+            low = minOrder.get(0).min;
+            high = maxOrder.get(maxOrder.size() - 1).max;
+
+            int totalPoints = minOrder.size() * 2;
+            int midIndex = totalPoints / 2;
+            int i = 0, j = 0, count = 0;
+            while (count < midIndex)
```

Comment:
this could be just `while (count < minOrder.size())` and remove `totalPoints` and `midIndex` above